omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ConditionController2_2Test.java

```diff
@@ -161,6 +163,80 @@ public void shouldCreateACodedCondition() throws Exception {
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("condition", codedOrFreeText);
+		attributes.add("clinicalStatus", ConditionClinicalStatus.ACTIVE);
+		attributes.add("verificationStatus", ConditionVerificationStatus.CONFIRMED);
+		attributes.add("onsetDate", "2018-02-23 00:00:00");
+		attributes.add("voided", false);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/"
+		        + RestTestConstants2_2.NON_CODED_CONDITION_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		condition = conditionService.getConditionByUuid(RestTestConstants2_2.NON_CODED_CONDITION_UUID);
+		
+		LinkedHashMap cond = attributes.get("condition");
+		
+		Assert.assertNotNull(condition);
+		Assert.assertNotEquals(oldCondition.getNonCoded(), condition.getCondition().getNonCoded());
+		Assert.assertEquals(cond.get("nonCoded"), condition.getCondition().getNonCoded());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, condition.getClinicalStatus());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, condition.getVerificationStatus());
+		Assert.assertNotNull(condition.getOnsetDate());
+	}
+	
+	@Test
+	public void shouldUpdateACodedCondition() throws Exception {
+		Condition condition = conditionService.getConditionByUuid(RestTestConstants2_2.CODED_CONDITION_UUID);
+		CodedOrFreeText oldCondition = condition.getCondition();
+		
+		Assert.assertNotNull(condition);
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
+		codedOrFreeText.add("coded", concept2.getUuid());
+		codedOrFreeText.add("specificName", conceptName2.getUuid());
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("condition", codedOrFreeText);
+		attributes.add("clinicalStatus", ConditionClinicalStatus.ACTIVE);
+		attributes.add("verificationStatus", ConditionVerificationStatus.CONFIRMED);
+		attributes.add("onsetDate", "2018-02-23 00:00:00");
+		attributes.add("voided", false);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + RestTestConstants2_2.CODED_CONDITION_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		condition = conditionService.getConditionByUuid(RestTestConstants2_2.CODED_CONDITION_UUID);
+		
+		LinkedHashMap cond = attributes.get("condition");
+		
+		Assert.assertNotNull(condition);
+		Assert.assertNotEquals(oldCondition.getCoded(), condition.getCondition().getCoded());
+		Assert.assertNotEquals(oldCondition.getSpecificName(), condition.getCondition().getSpecificName());
+		Assert.assertEquals(cond.get("coded"), condition.getCondition().getCoded().getUuid());
+		Assert.assertEquals(cond.get("specificName"), condition.getCondition().getSpecificName().getUuid());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, condition.getClinicalStatus());
```

Comment:
How sure are we that these did not have the same values before editing?

Reply:
Based on how I've done it, we can't be sure. Let me rectify that.